Content.Shared/Paper/PaperSystem.cs

```diff
@@ -131,10 +132,17 @@ private void OnInteractUsing(Entity<PaperComponent> entity, ref InteractUsingEve
                 var writeEvent = new PaperWriteEvent(entity, args.User);
                 RaiseLocalEvent(args.Used, ref writeEvent);
 
-                entity.Comp.Mode = PaperAction.Write;
                 _uiSystem.OpenUi(entity.Owner, PaperUiKey.Key, args.User);
                 UpdateUserInterface(entity);
+                if (_net.IsServer)
+                {
+                    var toolNetEnt = EntityManager.GetNetEntity(args.Used);
+                    _uiSystem.ServerSendUiMessage(entity.Owner, PaperUiKey.Key, new PaperBeginEditMessage(toolNetEnt), args.User);
+                }
```

Comment:
... Isn't this just an OpenUI event? Why is it even being networked?

Reply:
There's two UI events happening here - an OpenUI event which, for players reading the paper, updates the UI with new text (or opens the UI for players beginning the interaction).

This new message is sent only to the player who is attempting to edit the paper, and enables the edit controls. This separation is to fix the bug in the PR description[0]. I've added some comments explaining what's happening.

[0] This one: "If PlayerA was reading a paper and PlayerB started editing it, PlayerA would be shown the edit UI and would be able to write the paper without a pen."